lib/compress/zstd_ldm.c

```diff
@@ -41,12 +50,22 @@ void ZSTD_ldm_adjustParameters(ldmParams_t* params, U32 windowLog)
     params->bucketSizeLog = MIN(params->bucketSizeLog, params->hashLog);
 }
 
-size_t ZSTD_ldm_getTableSize(U32 hashLog, U32 bucketSizeLog) {
-    size_t const ldmHSize = ((size_t)1) << hashLog;
-    size_t const ldmBucketSizeLog = MIN(bucketSizeLog, hashLog);
+size_t ZSTD_ldm_getTableSize(ldmParams_t params, size_t maxChunkSize)
+{
+    size_t const ldmHSize = ((size_t)1) << params.hashLog;
+    size_t const ldmBucketSizeLog = MIN(params.bucketSizeLog, params.hashLog);
     size_t const ldmBucketSize =
-        ((size_t)1) << (hashLog - ldmBucketSizeLog);
-    return ldmBucketSize + (ldmHSize * (sizeof(ldmEntry_t)));
+        ((size_t)1) << (params.hashLog - ldmBucketSizeLog);
+    size_t const maxNbSeq =
+        ZSTD_ldm_getMaxNbSeq(maxChunkSize, params.minMatchLength);
+    size_t const totalSize = ldmBucketSize + ldmHSize * sizeof(ldmEntry_t) +
+                             maxNbSeq * sizeof(rawSeq);
+    return params.enableLdm ? totalSize : 0;
+}
+
+size_t ZSTD_ldm_getMaxNbSeq(size_t maxChunkSize, U32 minMatchLength)
+{
+    return maxChunkSize / minMatchLength;
```

Comment:
What's the value of `minMatchLength` when `params.enableLdm==0` ?
Maybe add `assert(minMatchLength > 0)` too ?